ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFDateFormat.java

```diff
@@ -51,18 +50,17 @@
  */
 @Description(name = "date_format", value = "_FUNC_(date/timestamp/string, fmt) - converts a date/timestamp/string "
     + "to a value of string in the format specified by the date format fmt.",
-    extended = "Supported formats are SimpleDateFormat formats - "
-        + "https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html. "
+    extended = "Supported formats are DateTimeFormatter formats - "
+        + "https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html. "
```

Comment:
I have verified that all the previous test cases work except following
1. `S` gives fraction-of-second instead of milliseconds. (for milliseconds you need `SSS` instead of `S`)
2. `u` gives year instead of Day number of week (1 = Monday, ..., 7 = Sunday). (Use e in newer format)

Reply:
@zabetak I think, if these 2 are the only change, then we can go ahead with DateTimeFormatter. What is your opinion?

Reply:
Personally, I am in favor of using the new API. However, I am sure that a user/client will discover that they get different results and may come back to us complaining about it. Still I think it is a good idea to use the new `DateTimeFormatter` so I would suggest to clearly indicate this small change of behavior in the JIRA and move forward.

Other than that since we are enhancing the API of `date_format` we should probably consider adding a few more tests in `TestGenericUDFDateFormat` with the supported patterns (could be done in a follow up I guess).